File: Server/src/main/java/example/example/ConcurrentServer.java
package example.example;

import java.net.Socket;

public abstract class ConcurrentServer extends Server{
    public ConcurrentServer(int port) {
        super(port);
    }

    protected void processRequest(Socket client) throws Exception {
        Thread worker = createWorker(client);
        worker.start();
    }

    protected abstract Thread createWorker(Socket client) throws Exception;
}

File: Server/ConcurrentServer.cs
using System.Net.Sockets;

namespace Server;

public abstract class ConcurrentServer : IServer
{
    protected ConcurrentServer(int port, string host) : base(port, host)
    { }

    public override void ProcessRequest(TcpClient client)
    {
        Thread thread = CreateWorker(client);
        thread.Start();
    }
    
    protected abstract Thread CreateWorker(TcpClient client);
}